server/storage.go

```diff
@@ -561,3 +564,124 @@ func saveGDriveToken(path string, token *oauth2.Token, logger *log.Logger) {
 
 	json.NewEncoder(f).Encode(token)
 }
+
+type StorjStorage struct {
+	Storage
+	project *uplink.Project
+	bucket  *uplink.Bucket
+	logger  *log.Logger
+}
+
+func NewStorjStorage(access, bucket string, logger *log.Logger) (*StorjStorage, error) {
+	var instance StorjStorage
+	var err error
+
+	ctx := context.TODO()
+
+	parsedAccess, err := uplink.ParseAccess(access)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.project, err = uplink.OpenProject(ctx, parsedAccess)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.bucket, err = instance.project.EnsureBucket(ctx, bucket)
+	if err != nil {
+		//Ignoring the error to return the one that occurred first, but try to clean up.
+		_ = instance.project.Close()
+		return nil, err
+	}
+
+	instance.logger = logger
+
+	return &instance, nil
+}
+
+func (s *StorjStorage) Type() string {
+	return "storj"
+}
+
+func (s *StorjStorage) Head(token string, filename string) (contentType string, contentLength uint64, err error) {
+	key := storj.JoinPaths(token, filename)
+
+	ctx := context.TODO()
+
+	obj, err := s.project.StatObject(ctx, s.bucket.Name, key)
+	if err != nil {
+		return "", 0, err
+	}
+
+	customMeta := obj.Custom.Clone()
+	contentType = customMeta["contentType"]
```

Comment:
MinIO uses "content-type" (not "contentType") as the key in metadata. Should it be the same here?